Blocstagram/Blocstagram/MediaTableViewCell.m

```diff
@@ -0,0 +1,232 @@
+                                                                                     constant:100];
+        self.usernameAndCaptionLabelHeightConstraint.identifier = @"Username and caption label height constraint";
+        
+        self.commentLabelHeightConstraint = [NSLayoutConstraint constraintWithItem:_commentLabel
+                                                                         attribute:NSLayoutAttributeHeight
+                                                                         relatedBy:NSLayoutRelationEqual
+                                                                            toItem:nil
+                                                                         attribute:NSLayoutAttributeNotAnAttribute
+                                                                        multiplier:1
+                                                                          constant:100];
+        self.commentLabelHeightConstraint.identifier = @"Comment label height constraint";
+        
+        [self.contentView addConstraints:@[self.usernameAndCaptionLabelHeightConstraint,
+                                           self.commentLabelHeightConstraint]];
+    }
+    
+    return self;
+}
+
+#pragma mark - UITableCell specific
+
+- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
+    [super setSelected:selected animated:animated];
+}
+
+#pragma mark - Style
+
+- (void)layoutSubviews {
+    [super layoutSubviews];
+    
+    CGSize maxSize = CGSizeMake(CGRectGetWidth(self.bounds), CGFLOAT_MAX);
+    CGSize usernameLabelSize = [self.usernameAndCaptionLabel sizeThatFits:maxSize];
+    CGSize commentLabelSize = [self.commentLabel sizeThatFits:maxSize];
+    
+    // customing height for constraints
+    self.usernameAndCaptionLabelHeightConstraint.constant = usernameLabelSize.height + 20;
+    self.commentLabelHeightConstraint.constant = commentLabelSize.height + 20;
+    
+    /* retain aspect ratio */
+    CGFloat ratio = self.media.image.size.width / self.media.image.size.height;
+    NSLayoutConstraint *aspectRatioConstraint = [NSLayoutConstraint constraintWithItem:self.mediaImageView
+                                                                             attribute:NSLayoutAttributeWidth
+                                                                             relatedBy:NSLayoutRelationEqual
+                                                                                toItem:self.mediaImageView
+                                                                             attribute:NSLayoutAttributeHeight
+                                                                            multiplier:ratio
+                                                                              constant:0.0f];
+    [self.contentView addConstraint:aspectRatioConstraint];
+    
+    NSLog(@"%f, %f", self.mediaImageView.frame.size.width, self.mediaImageView.frame.size.height);
```

Comment:
One thing I would do, when debugging this missing image issue, is give myself some useful info from each log... so, for example, I would want to see maybe how often this code is getting run for each media item, so my log might look more like this:
```
    NSLog(@"mediaImage [%p] view:%f, %f", self.media, self.mediaImageView.frame.size.width, self.mediaImageView.frame.size.height);
```

Printing out the address (the `%p`) will let me know which image I'm looking at the dimensions for... I might even add the `[NSThread callStackSymbols]` like we talked about on Friday... but that tends to dump a lot to the console, so I would probably just set a breakpoint in the method near the log line and step through it a few times to see where we are, in the stack. We can walk through the debugging facilities in Xcode on Friday, but a great starter set of URLs to look through are Apple's documentation (https://developer.apple.com/library/ios/recipes/xcode_help-debugger/articles/about_debug_area.html#//apple_ref/doc/uid/TP40009986-CH3-SW1 and https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/gdb_to_lldb_transition_guide/document/Introduction.html) and this 3rd party developer's blog post about using LLDB inside Xcode (http://devmonologue.com/ios/ios/using-the-debugger-console-lldb-in-xcode/).